code/game/objects/structures/tables_racks.dm

```diff
@@ -35,6 +35,29 @@
 	var/deconstruction_ready = TRUE
 	var/flipped = 0
 
+/obj/structure/table/Bumped(mob/living/carbon/human/H)
+	. = ..()
+	if(!istype(H) || H.shoes || H.stat == DEAD || istype(H.dna.species, /datum/species/shadow/ling))
```

Comment:
Why shadowlings, and only shadowlings? Also, why == DEAD and not != CONSCIOUS?

Reply:
I believe shadowlings has a exoskeleton and aren't affected by glass shards and the such, since this checks for shoes and not your species, shadowlings need this since they have inherent glass shard immunity.